internal/k8s/reconciler/route.go

```diff
@@ -135,7 +136,7 @@ func (r *K8sRoute) SyncStatus(ctx context.Context) error {
 	return nil
 }
 
-func (r *K8sRoute) Resolve(listener store.Listener) *core.ResolvedRoute {
+func (r *K8sRoute) Resolve(listener store.Listener) core.ResolvedRoute {
```

Comment:
Small signature change since everywhere that was consuming this function was dereferencing the `ResolvedRoute` anyways